Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/PersonStats.hs

```diff
@@ -0,0 +1,158 @@
+  latestSearch <- runInReplica $ QSR.findLatestSearchRequest personId
+  decEmail <- decrypt `mapM` person.email
+  emergencyContactsNum <- length <$> runInReplica (QPDEN.findAllByPersonId personId)
+  let userCancellationRate =
+        fromIntegral personStats.userCancelledRides
+          / fromIntegral (personStats.userCancelledRides + personStats.completedRides)
+      overalCancellationRate =
+        fromIntegral (personStats.userCancelledRides + personStats.driverCancelledRides)
+          / fromIntegral (personStats.userCancelledRides + personStats.driverCancelledRides + personStats.completedRides)
+      weekdayMorningPeakRidesRate =
+        fromIntegral personStats.morningPeakRides
+          / fromIntegral personStats.completedRides
+      weekdayEveningPeakRidesRate =
+        fromIntegral personStats.eveningPeakRides
+          / fromIntegral personStats.completedRides
+      offPeakRidesRate =
+        fromIntegral personStats.offPeakRides
+          / fromIntegral personStats.completedRides
+      weekendRidesRate =
+        fromIntegral personStats.weekendRides
+          / fromIntegral personStats.completedRides
+      weekdayRidesRate =
+        fromIntegral personStats.weekdayRides
+          / fromIntegral personStats.completedRides
+      weekendPeakRideRate =
+        fromIntegral personStats.weekendPeakRides
+          / fromIntegral personStats.completedRides
+      daysFromRegDate :: Int = roundToIntegral $ diffUTCTime now person.createdAt / 60 / 60 / 24
+      avgRidesPerWeek :: Float = fromIntegral completedRidesCount / (fromIntegral daysFromRegDate / 7)
+
+  return $
+    PersonStatsRes
+      { riderId = person.id,
+        signupDate = person.createdAt,
+        email = decEmail,
+        isWhatsAppOptInStatus = Just WhatsApp.OPT_IN == person.whatsappNotificationEnrollStatus,
+        lifetimeRides = completedRidesCount,
+        lastRideTaken = latestCompletedRide <&> (.createdAt),
+        frequencyCategory = getFrequencyCategory completedRidesInLast30DaysCount,
+        isChurnedUser = maybe False (isChurned now) latestCompletedRide,
+        userCategory = getUserCategory avgRidesPerWeek,
+        isBlocked = person.blocked,
+        latestSearch = latestSearch <&> (.createdAt),
+        latestSearchFrom = latestSearch <&> getCoordinates . (.fromLocation),
+        favoriteLocationsNum = favLocNum,
+        commonAppUseCase = getCommonAppUseCase weekdayRidesRate weekendRidesRate completedRidesCount avgRidesPerWeek,
+        ..
+      }
+  where
+    days30 :: NominalDiffTime = 30 * 24 * 60 * 60
+    isChurned now latestCompletedRide = do
+      diffUTCTime now latestCompletedRide.createdAt > days30
+    getCommonAppUseCase weekdayRidesRate weekendRidesRate completedRidesCount avgRidesPerWeek
+      | weekdayRidesRate > 0.6 && completedRidesCount > 4 = WEEKDAY
+      | weekendRidesRate > 0.6 && completedRidesCount > 4 = WEEKEND
```

Comment:
So WEEKDAY is priority even if weekend rate is high?

Reply:
yeah

Reply:
But why? We can have one `DAILY` enum also

Reply:
as discussed, we can leave this as it is